### PR TITLE
Resource managers can generate a utilisation report

### PR DESCRIPTION
**Pull request by AndrewVos**

### CODE REVIEW

File: spec/factories/bookable_slots.rb [hunk 1]
```diff
@@ -2,5 +2,6 @@
   factory :bookable_slot do
     start_at Time.zone.now
     end_at { start_at + 1.hour }
+    guider { create(:user) }
```
Comment (benlovell) [inline spec/factories/bookable_slots.rb]: This should be `create(:guider)` or weird stuff can happen right?

File: spec/models/bookable_slot_spec.rb [hunk 2]
```diff
@@ -36,6 +36,91 @@ def result
...
+    end
+
+    it 'excludes slots with user holidays' do
+      create(
+        :holiday,
+        user: guider,
+        start_at: make_time(5, 30),
+        end_at: make_time(20, 30)
+      )
+
```
Comment (benlovell) [inline spec/models/bookable_slot_spec.rb]: Remove this line or add one elsewhere in this same style (actually my preference) to separate the setup from the assertion(s).

File: spec/models/bookable_slot_spec.rb [hunk 3]
```diff
@@ -36,6 +36,91 @@ def result
     end
   end
 
+  describe '#without_appointments' do
+    let(:guider) do
+      create(:user)
```
Comment (AndrewVos) [inline spec/models/bookable_slot_spec.rb]: done

File: app/forms/appointment_report.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  def generate
+    field = where.to_sym
+    appointments = Appointment.all.order(field)
+    appointments = appointments.where(field => range) if range
+    CSV.generate do |csv|
+      csv << EXPORTABLE_ATTRIBUTES
+      appointments.each do |appointment|
+        presenter = AppointmentCsvPresenter.new(appointment)
+        csv << EXPORTABLE_ATTRIBUTES.map { |a| presenter.send(a) }
```
Comment (benlovell) [inline app/forms/appointment_report.rb]: `public_send` is safer.

File: app/forms/utilisation_report.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
+require 'csv'
+
+class UtilisationReport
+  include ActiveModel::Model
+  include Report
+
+  attr_reader :date_range
+
+  def initialize(params = {})
+    @date_range = params[:date_range]
```
Comment (benlovell) [inline app/forms/utilisation_report.rb]: Same here with using `fetch` to be more defensive.

File: spec/features/resource_manager_downloads_utilisation_reports_spec.rb [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  end
+
+  let(:date_range) do
+    [
+      I18n.l(start_at, format: :date_range_picker),
+      I18n.l(end_at, format: :date_range_picker)
+    ].join(' - ')
+  end
+
+  # rubocop:disable Metrics/MethodLength
```
Comment (benlovell) [inline spec/features/resource_manager_downloads_utilisation_reports_spec.rb]: You can comma separate these rules onto one line.

File: app/forms/appointment_report.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    :date_of_birth,
+    :booking_reference,
+    :memorable_word,
+    :phone,
+    :mobile,
+    :email
+  ].freeze
+
+  def initialize(params = {})
+    @where = params[:where]
```
Comment (AndrewVos) [inline app/forms/appointment_report.rb]: done

File: app/forms/appointment_report.rb (hunk 4 above)
Comment (AndrewVos) [inline app/forms/appointment_report.rb]: done

**Pull request merged**